gc/src/lib.rs

```diff
@@ -121,6 +122,76 @@ impl<T: Trace + ?Sized> Gc<T> {
     }
 }
 
+impl<T: Trace + ?Sized> Gc<T> {
+    /// Consumes the `Gc`, returning the wrapped pointer.
+    /// 
+    /// To avoid a memory leak, the pointer must be converted back into a `Gc`
+    /// using [`Gc::from_raw`][from_raw].
+    /// 
+    /// [from_raw]: struct.Gc.html#method.from_raw
+    /// 
+    /// # Examples
+    /// 
+    /// ```
+    /// use gc::Gc;
+    /// 
+    /// let x = Gc::new(22);
+    /// let x_ptr = Gc::into_raw(x);
+    /// assert_eq!(unsafe { *x_ptr }, 22);
+    /// ```
+    pub fn into_raw(this: Self) -> *const T {
+        let ptr: *const T = &*this;
+        mem::forget(this);
+        ptr
+    }
+
+    /// Constructs an `Gc` from a raw pointer.
+    ///
+    /// The raw pointer must have been previously returned by a call to a
+    /// [`Gc::into_raw`][into_raw].
+    ///
+    /// This function is unsafe because improper use may lead to memory
+    /// problems. For example, a double-free may occur if the function is called
```

Comment:
I don't know if a double-free would necessarily occur. The more likely problem would be a use-after-free.